### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by kianmeng**
Besides other documentation changes, this commit ensures the generated
HTML doc for HexDocs.pm will become the source of truth for this Elixir
library and leverage on latest features of ExDoc.

### CODE REVIEW

**Comment by zookzook** [comment]
Many thanks for your efforts. I will review the PR as soon as possible.

File: .gitignore [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
+# Also ignore archive artifacts (built via "mix archive.build").
 *.ez
-tmp/
-.tool-versions
-.elixir_ls
-plt_core_path/*.plt
-.idea/
-*.iml
-*.code-workspace
-data/
```
Comment (zookzook) [inline .gitignore:13]: Why do you removed this patterns?

File: lib/mongo/bulk_write.ex [hunk 2]
```diff
@@ -399,7 +399,7 @@ defmodule Mongo.BulkWrite do
 
     [q: filter,
      limit: Keyword.get(opts, :limit),
-     collation: Keyword.get(opts, :collaction)] |> filter_nils()
+     collation: Keyword.get(opts, :collection)] |> filter_nils()
```
Comment (zookzook) [inline lib/mongo/bulk_write.ex:402]: I guess, this was accidentally changed. The option key is `collation`.

**Review by zookzook — CHANGES_REQUESTED** [review]
Very nice work. I just added two comments. Maybe you could provide some feedback.

File: .gitignore (hunk 1 above)
Comment (kianmeng) [reply .gitignore]: Added back.

File: lib/mongo/bulk_write.ex (hunk 2 above)
Comment (kianmeng) [reply lib/mongo/bulk_write.ex]: Fixed.

Comment (kianmeng) [reply lib/mongo/bulk_write.ex]: Fixed.

**Review by zookzook — APPROVED** [review]
Thank you!

**Pull request merged**